docs/guide/core-concepts.rst

```diff
@@ -558,435 +576,592 @@ possible, Constant ``NoData`` values are to be preferred. For
+   :alt:
+
+The ``Tile`` trait has operations you'd expect for traversing and
+transforming the contents of the tiles, like:
+
+-  ``map: (Int => Int) => Tile``
+-  ``foreach: (Int => Unit) => Unit``
+-  ``combine: Tile => ((Int, Int) => Int) => Tile``
+-  ``color: ColorMap => Tile``
+
+As discussed above, the ``Tile`` interface carries information about how big
+it is and what its underlying `Cell Type <#cell-types>`__ is:
+
+-  ``cols: Int``
+-  ``rows: Int``
+-  ``cellType: CellType``
 
 .. raw:: html
 
    <hr>
 
-Raster Data
-===========
 
-    “Yes raster is faster, but raster is vaster and vector just SEEMS
-    more corrector.” — `C. Dana
-    Tomlin <http://uregina.ca/piwowarj/NotableQuotables.html>`__
+Layouts and Tile Layers
+=======================
+
+The core vector and raster functionality thus far described stands on its own
+for small scale applications.  But, as mentioned, Geotrellis is intended to
+work with big data in a distributed context.  For this, we rely on Apache
+Spark's resilient distributed dataset (RDD).  RDDs of both raster and vector
+data are naturally supported by Geotrellis, but some new concepts are required
+to integrate this abstraction for distributed processing.
+
+For most applications, the data of interest must be keyed to a layout to give
+the content of an RDD—which is usually a collection of key-value pairs (i.e.,
+``RDD[(K, V)]``)—a consistent interpretation as a cohesive raster.  In such an
+RDD, the key type, ``K``, is one of ``TemporalKey``, ``SpatialKey``, or
+``SpaceTimeKey``.  The latter two key types obviously contain spatial data
+(declared in context bounds as ``[K: SpatialComponent]``, where values of such
+a type ``K`` can have their spatial component extracted using the
+``getComponent[SpatialKey]`` extension method), which is used to identify a
+region in space.
+
+The |LayoutDefinition|_ class is used to describe how
+``SpatialKey``\ s map to regions in space.  The ``LayoutDefinition`` is a
+``GridExtent`` subclass defined with an ``Extent`` and ``CellSize``.  The
+``Extent`` is subdivided into a grid of uniform, rectangular regions.  The
+size and number of the sub-regions is determined using the ``CellSize`` of the
+``LayoutDefinition``, and then the pixel dimensions of the constituent tiles.
+The sub-regions are then assigned a ``SpatialKey`` with the ``(0, 0)``
+position corresponding to the upper-left corner of the extent; the `x`
+coordinate increases toward the right, and the `y` coordinate increases moving
+down (into lower latitude values, say).
+
+.. |LayoutDefinition| replace:: ``geotrellis.spark.tiling.LayoutDefinition``
+.. _LayoutDefinition: https://geotrellis.github.io/scaladocs/latest/#geotrellis.spark.tiling.LayoutDefinition
+
+Thus far, we've described how an ``RDD[(K, V)]`` plus a ``LayoutDefinition``
+can be used to represent a large, distributed raster (when ``[K:
+SpatialComponent]``).  To solidify this notion, Geotrellis has a notion of a
```

Comment:
Like with `LayoutDefinition`, having a subsection for `Metadata[M]` might be useful.

Reply:
Eh.  `Metadata` is kind of a nothing class that just means that an implementation has a `metadata` field of type `M`.  I don't want this document to balloon up by having sections for every little thing.